oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/authentication/ReactiveOidcIdTokenDecoderFactory.java

```diff
@@ -61,17 +70,55 @@
 			put(MacAlgorithm.HS512, "HmacSHA512");
 		}
 	};
+	private static final Converter<Map<String, Object>, Map<String, Object>> DEFAULT_CLAIM_TYPE_CONVERTER =
```

Comment:
Thinking out loud a bit, but this API doesn't account for the fact that conversion may be a blocking operation. For example, if we have an `InputStream` that needs converted to an Object (i.e. reading in JSON) that is blocking and the API does not return a reactive type.

I'm wondering if in practice this is a concern in this situation since we have likely already done all the IO (it probably is converting from a String not an `InputStream`). What do we think the likelihood is that we will need to do a blocking operation during conversion? Are there other reasons we might need to block during conversion?

Reply:
@rwinch 

> I'm wondering if in practice this is a concern in this situation since we have likely already done all the IO 

The `source` of claims passed into the `Converter` should have already been parsed from it's JSON representation. For the ID Token use case, Nimbus's JSON parser would have read/parsed all claims in the raw ID Token (including custom claims) into an instance of `JWTClaimsSet`. Even for cases where there is a custom claim and it's a JSON object, Nimbus would parse it into a `Map`.

> Are there other reasons we might need to block during conversion?

At this point, I do not see  a use case where a blocking operation could happen with the default `NimbusReactiveJwtDecoder`.